src/charm.py

```diff
@@ -8,7 +8,7 @@
 
 import json
 import logging
-import re
+from urllib.parse import urlparse
```

Comment:
This approach has been tried and found wanting. See commit entitled "Improve validation of target address" and commit message therein.

Reply:
Not sure what you mean. This seems to work well. Could you take a look at my recent changes?

Reply:
Excellent work. I particularly like the RFC comment. Just re-request review when you are done.